src/thor/route_action.cc

```diff
@@ -80,6 +80,42 @@ void via_discontinuity(
   }
 }
 
+inline bool is_through_point(const valhalla::Location& l) {
+  return l.type() == valhalla::Location::kThrough || l.type() == valhalla::Location::kBreakThrough;
+}
+
+inline bool is_break_point(const valhalla::Location& l) {
+  return l.type() == valhalla::Location::kBreak || l.type() == valhalla::Location::kBreakThrough;
+}
+
+inline bool is_high_reachable(const valhalla::Location& loc,
+                              const valhalla::Location_PathEdge& edge) {
+  return edge.inbound_reach() >= loc.minimum_reachability() &&
+         edge.outbound_reach() >= loc.minimum_reachability();
+}
+
+inline void remove_edges_with_low_reachability(valhalla::Location& loc) {
+  auto new_end = std::remove_if(loc.mutable_path_edges()->begin(), loc.mutable_path_edges()->end(),
+                                [&loc](const auto& pe) { return !is_high_reachable(loc, pe); });
+  int start_idx = std::distance(loc.mutable_path_edges()->begin(), new_end);
+  loc.mutable_path_edges()->DeleteSubrange(start_idx, loc.path_edges_size() - start_idx);
+
+  new_end = std::remove_if(loc.mutable_filtered_edges()->begin(), loc.mutable_filtered_edges()->end(),
+                           [&loc](const auto& pe) { return !is_high_reachable(loc, pe); });
+  start_idx = std::distance(loc.mutable_filtered_edges()->begin(), new_end);
+  loc.mutable_filtered_edges()->DeleteSubrange(start_idx, loc.filtered_edges_size() - start_idx);
+}
+
+inline void remove_all_edges_except_one(valhalla::Location& loc, const GraphId& edgeid) {
```

Comment:
if its a special case node snap, should allow both the edge ending at the node and the edge beginning at the node?

Reply:
i see this is just copy paste from below with also removing the filtered edges. best to keep it the same to avoid any unforeseen issues

Reply:
> i see this is just copy paste from below with also removing the filtered edges. best to keep it the same to avoid any unforeseen issues

agree, will replace with one function.

> if its a special case node snap, should allow both the edge ending at the node and the edge beginning at the node?

why is it important ? if we choose some edge and location has type `through`, we should be able to go through this edge